BUMP_VERSION.asciidoc

```diff
@@ -0,0 +1,196 @@
+-------------------------------------
+  -elasticsearch     = 5.2.0
+  +elasticsearch     = 5.3.0
+-------------------------------------
+
+- In `docs/groovy-api/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/java-api/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/java-rest/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/plugins/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+- In `docs/reference/index.asciidoc`, update the `version` to 5.3.0:
+-------------------------------------
+  -:version: 5.2.0
+  +:version: 5.3.0
+-------------------------------------
+
+==== Update the rolling-upgrade test version
+
+In the master branch, update the `qa/rolling-upgrade/build.gradle` by setting the `bwcVersion` 
+of the rolling upgrade tests to `5.3.0-SNAPSHOT`.  This should go away soon as we create a matrix 
+of backward compatibility versions to run the rolling upgrade tests against, and the versions will 
+be generated automatically.
+-------------------------------------
+  -bwcVersion = '5.2.0-SNAPSHOT'
+  +bwcVersion = '5.3.0-SNAPSHOT'
+-------------------------------------
+
+Before the rolling upgrade tests can be run, however, make sure the 5.3.0-SNAPSHOT build is already
+published to the staging artifacts (there is an internal CI job to generate the 5.x snapshot build).
+This should also go away soon, as work is underway to not require pulling down a snapshot build of
+the backward compatibility version but instead, the rolling upgrade test will checkout the backward
+compatibility version in question to a git worktree.
+
+
+== Steps to take after releasing
+
+After releasing a version of Elasticsearch (in our scenario, 5.2.0), then the following must 
+be done to bump to the next version of Elasticsearch.  Note that the same procedure is followed
+when bumping the bug-fix version (for example, after releasing 5.2.1, bumping the release version
+to 5.2.2).
+
+==== Upgrade the version
+
+Since 5.2.0 has been released, the next bug-fix version in the 5.2.0 line 
+will be 5.2.1.  Create a V_5_2_1 version constant in `core/src/main/java/org/elasticsearch/Version.java`
+for all branches >= 5.2.  In this case, the V_5_2_1 version constant should go into
+the 5.2, 5.x, and master branches.  
+
+For the 5.2 branch only, set the CURRENT version in `Version.java` to the new V_5_2_1 constant.  
+For the 5.x and master branches, just add the V_5_2_1 constant without setting the CURRENT version.
```

Comment:
I guess here, I was just trying to refer to the constant generally as the v5.2.1 constant, and in the example, I use UNRELEASED.  But I can change it.  Or I can refer to the constant instead as the 5.2.1 constant instead of the V_5_2_1 constant, to avoid any imbiguity.